plugins/inventory/vmware_vm_inventory.py

```diff
@@ -69,67 +69,69 @@
             description:
             - Specify the list of VMware schema properties associated with the VM.
             - These properties will be populated in hostvars of the given VM.
-            - Each value in the list specifies the path to a specific property in VM object.
+            - Each value in the list can be a path to a specific property in VM object or a path to a collection of VM objects.
+            - Set value to 'all' to query all properties.
+            - For all properties refer following URL
+            - See U(https://github.com/monkey-mas/lab/blob/master/pyvmomi/docs/vim/VirtualMachine.rst#attributes).
             type: list
             default: [ 'name', 'config.cpuHotAddEnabled', 'config.cpuHotRemoveEnabled',
                        'config.instanceUuid', 'config.hardware.numCPU', 'config.template',
-                       'config.name', 'guest.hostName', 'guest.ipAddress',
+                       'config.name', 'config.uuid' ,'guest.hostName', 'guest.ipAddress',
                        'guest.guestId', 'guest.guestState', 'runtime.maxMemoryUsage',
-                       'customValue'
+                       'customValue', 'summary.runtime.powerState', 'config.guestId'
                        ]
             version_added: "2.9"
-        datacenters:
+        resources:
```

Comment:
Yes, `resources` option replaced all of `datacenters`, `clusters`, `esxi_hostsystems`, `resource_poosl` and `folders` to a more flexible approach and virtually allow for any VIM types. This enable for a tree-like structure modeling which is native to vCenter users. Also, it is explicitly define the relation between those types.

For example to target VMs in `dev` folder of `Asia-Datacenter1` and `prod` folder of `Europe-Datacenter5`:
```yaml
resources:
- datacenter:
  - Asia-Datacenter1
  resources:
  - folder:
    - dev
- datacenter:
  - Europe-Datacenter5
  resources:
  - folder:
    - prod
```


Loosely speaking, this might not be a breaking change as previous options (`datacenters`...)  were recently added to this development repo and were not part of 2.9 release.

Reply:
I agree with @s-hertel, can we divide this PR into several PRs so that we can test it without any regression. Also, I also reverted my performance-related PR https://github.com/ansible-collections/vmware/pull/85 which is available in devel only.